src/pages/blog.js

```diff
@@ -1,20 +1,60 @@
 import React from 'react';
 import Layout from '../components/Layout';
+import {useState} from 'react';
+import JSONDATA from './test_data.json'
 
-const Blog = () => {
-  const BlogPost = () => {
-    return (
-      <div>
 
-      </div>
-    )
+//Added JSON dummy data to test the search bar.  
+const blogTitles = []
+JSONDATA.map((val,key) => {
+  blogTitles.push(val['Blog Title']);  
+})
+
+console.log(blogTitles)
+// filterBlogs takes in a term to filter for and a list of all blogs. 
+// It performs a case insensitive search on the title of the blog and returns
+// a list of blogs that match the filter term. If no blogs are found, it return nothing. 
+
+function filterBlogs(searchTerm, blogTitles){
+  
+    var filteredBlogTitles = []; 
+    for (var i = 0; i < blogTitles.length; i++){
+        if (blogTitles[i].toLowerCase().includes(searchTerm.toLowerCase())){
+            filteredBlogTitles.push(blogTitles[i]);
+        }
+    }
+    return filteredBlogTitles;
+}
+
+
+const Blog = () => {
+  
+  const searchItems = (searchValue) => {
+    setSearchInput(searchValue);
   }
+
+  const [searchInput, setSearchInput] = useState('');
+  var filteredBlogsList = filterBlogs(searchInput, blogTitles);
```

Comment:
You need to store the current blog posts selected in state.

1. You need some state variable representing the state of the blogs list.
`const [filteredBlogs, setFilteredBlogs] = useState(<<originalBlogList>>)`

2. You need to dynamically update the state depending on whether the search input state has changed.
`useEffect(() => {
  setFilteredBlogs(filterBlogs(searchInput)) 
}, [searchInput])`
** note: add searchInput as a dependency to useEffect so that it will be called on change.

Reply:
Used another state variable for blogs list but did not need second state for search input state.